### ISSUE TITLE
Snowpack Compile - The 'this' keyword is equivalent to 'undefined at the top level of an ES module

### ISSUE THREAD

**Issue by khanakia** [ant-design/ant-design#26090]
- [x] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### Reproduction link
[https://github.com/khanakia/snowpack-antd-issue.git](https://github.com/khanakia/snowpack-antd-issue.git)

### Steps to reproduce
git clone https://github.com/khanakia/snowpack-antd-issue.git
yarn install
yarn start --reload

ERROR Screen
https://prnt.sc/tvumbi

### What is expected?
Should not give any error

### What is actually happening?
Giving error while compiling [warn] The 'this' keyword is equivalent to 'undefined' at the top level of an ES module, and has been rewritten

| Environment | Info |
|---|---|
| antd | 4.5.2 |
| React | 16.13.1 |
| System | OSX |
| Browser | Chrome |

---
Check screenshot please https://prnt.sc/tvumbi
I am using Snowpack https://www.snowpack.dev/#get-started

**Comment by afc163** [comment]
antd don't work with Snowpack now, feel free to send us PR.

**Comment by snaerth** [comment]
@afc163 do you know what we need to do? I would be happy to help because I am having the same problem

**Comment by afc163** [comment]
@snaerth I know litter about snowpack, could you help to dig and resolve it?

**Comment by afc163** [comment]
Another snowpack issue: #26226

@sam-chuang 

---

- [x] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### Reproduction link
[https://github.com/sam-chuang/ant-design-snowpack-issue](https://github.com/sam-chuang/ant-design-snowpack-issue)

### Steps to reproduce
1. clone https://github.com/sam-chuang/ant-design-snowpack-issue
2. npm install & npm start
3. open http://localhost:8080/ on browser
4. open browser dev console

### What is expected?
Page console should not have any warning

Warning: React.createElement: type is invalid

### What is actually happening?
Page console shows multiple warning about React.createElement: type is invalid

Refer to screenshots here
https://sam-chuang.github.io/ant-design-snowpack-issue/
![issue](https://user-images.githubusercontent.com/33828310/90325335-e81ff980-dfac-11ea-9a43-637eaed25170.png)


| Environment | Info |
|---|---|
| antd | 4.5.4 |
| React | 16.13.1 |
| System | Windows |
| Browser | Chrome |

**Comment by afc163** [comment]
Another snowpack issue: #26237

@sam-chuang 

---

- [x] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### Reproduction link
[https://github.com/sam-chuang/ant-design-table-pagination-snowpack-issue](https://github.com/sam-chuang/ant-design-table-pagination-snowpack-issue)

### Steps to reproduce
1. Clone https://github.com/sam-chuang/ant-design-table-pagination-snowpack-issue
2. npm install
3. npm start
4. open browser

### What is expected?
Show table and pagination correctly

### What is actually happening?
blank page, browser console shows errors

| Environment | Info |
|---|---|
| antd | 4.5.4 |
| React | 16.13.1 |
| System | Windows |
| Browser | Chrome |

**Comment by toneb** [comment]
I investigated the problems. #26090 is solved in snowpack 2.8.0.

While #26226 and #26237 both occur due to a problem in structure of antd icons es module. Icons should be in the root folder, as they are in source:
![image](https://user-images.githubusercontent.com/616279/90668219-03f6fa00-e250-11ea-974f-e405c0daa9cc.png)


But instead they are contained in icons subfolder:
![image](https://user-images.githubusercontent.com/616279/90668419-605a1980-e250-11ea-8077-395bebc65a63.png)


Workaround is to use snowpack alias for icons, which solves the issues for antd, but not for antd-pro, since antd-pro is importing types from index.js, which does not exist in icons subfolder.

WORKAROUND (doesn't work with antd-pro):
![image](https://user-images.githubusercontent.com/616279/90668732-cd6daf00-e250-11ea-8119-57de944fe3ef.png)

**Comment by toneb** [comment]
Another alternative would be to import icons differently in antd, currently they are imported as:
```import DoubleLeftOutlined from '@ant-design/icons/DoubleLeftOutlined';```

while alternative would be:
```import { DoubleLeftOutlined } from '@ant-design/icons;```
or 
```import DoubleLeftOutlined from '@ant-design/icons/icons/DoubleLeftOutlined';```.

Icons mainteiners should express their intent - how should the icons be imported ... are all three cases supported?

**Comment by toneb** [comment]
I further investigated the causes. Problem was in missing esm entry points for icons. I created PR https://github.com/ant-design/ant-design-icons/pull/356 which fixes issues #26226 and #26237.

While the issue in #26090 still remains, I was a little early to call it fixed in snowpack 2.8.0 ... I had some other parameters modified which improved the situation.

**Comment by khanakia** [comment]
@toneb so what's the solution ?

**Comment by toneb** [comment]
@khanakia I do not have a solution. I had a custom configuration which had hidden some of the warnings.

But, I did investigate what the root cause is, and found this: https://github.com/rollup/rollup/issues/794

The "problem" is in configuration of typescript which produces some polyfills with global this, example from button.js:
![image](https://user-images.githubusercontent.com/616279/91662175-e7c14b80-eae0-11ea-8fe9-e004955ccfbf.png)

I think the antd team should consider increasing the target from es6 to something newer (maybe only for esm builds), but at the end, this are only compilation warnings, I did not found them to be causing any actual problems.
